### PR TITLE
update gradle version to 8.1.1

### PR DESCRIPTION
**Pull request by MrManHD**

### CODE REVIEW

File: simplecloud-base/build.gradle [hunk 1]
```diff
@@ -24,44 +24,42 @@ plugins {
...
+    compileOnly group: 'org.reflections', name: 'reflections', version: '0.9.12'
+    compileOnly group: 'org.litote.kmongo', name: 'kmongo', version: dependencyKMongoVersion
+    compileOnly(group: 'io.netty', name: 'netty-all', version: dependencyNettyVersion)
+    compileOnly group: 'org.apache.commons', name: 'commons-lang3', version: '3.12.0'
 }
 
-configurations {
-    jar.archiveName = 'base.jar'
+jar {
+    archiveFileName.set("base-old.jar")
```
Comment (FllipEis) [inline simplecloud-base/build.gradle:54]: Can be removed

File: simplecloud-dependency-loader/build.gradle [hunk 2]
```diff
@@ -1,31 +1,27 @@
...
+    compileOnly 'org.eclipse.aether:aether-transport-http:1.1.0'
+    compileOnly 'org.apache.maven:maven-aether-provider:3.3.9'
+    compileOnly(project(":simplecloud-runner"))
+    compileOnly(project(":simplecloud-api"))
 }
 
-configurations {
-    jar.archiveName = 'dependency-loader.jar'
+jar {
+    archiveFileName.set("dependency-loader-old.jar")
```
Comment (FllipEis) [inline simplecloud-dependency-loader/build.gradle:13]: Can be removed

File: simplecloud-launcher/build.gradle [hunk 3]
```diff
@@ -21,43 +21,45 @@
...
+    testApi group: 'org.jetbrains.kotlinx', name: 'kotlinx-coroutines-core', version: dependencyKotlinCouroutinesVersion
+    testApi group: 'com.google.code.gson', name: 'gson', version: '2.9.0'
+    testApi group: 'org.apache.commons', name: 'commons-lang3', version: '3.12.0'
 
 }
 
-configurations {
-    jar.archiveName = 'launcher.jar'
+jar {
+    archiveFileName.set("launcher-old.jar")
```
Comment (FllipEis) [inline simplecloud-launcher/build.gradle:58]: Can be removed

File: simplecloud-launcher/src/test/kotlin/eu/thesimplecloud/launcher/external/ModuleListLoaderTest.kt
Comment (FllipEis) [inline simplecloud-launcher/src/test/kotlin/eu/thesimplecloud/launcher/external/ModuleListLoaderTest.kt:1]: Why was this deleted?

File: simplecloud-modules/simplecloud-module-chat-tab/build.gradle [hunk 4]
```diff
@@ -20,18 +20,22 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-configurations {
-    jar.archiveName = 'SimpleCloud-Chat+Tab.jar'
+jar {
+    archiveFileName.set("SimpleCloud-Chat+Tab-old.jar")
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-chat-tab/build.gradle:24]: Can be removed

File: simplecloud-modules/simplecloud-module-cloudflare/build.gradle [hunk 5]
```diff
@@ -20,15 +20,19 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-configurations {
-    jar.archiveName = 'SimpleCloud-CloudFlare.jar'
+jar {
+    archiveFileName.set('SimpleCloud-CloudFlare-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-cloudflare/build.gradle:24]: Can be removed

File: simplecloud-modules/simplecloud-module-hubcommand/build.gradle [hunk 6]
```diff
@@ -20,14 +20,17 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-configurations {
-    jar.archiveName = 'SimpleCloud-HubCommand.jar'
+jar {
+    archiveFileName.set('SimpleCloud-HubCommand-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-hubcommand/build.gradle:24]: Can be removed

File: simplecloud-modules/simplecloud-module-internalwrapper/build.gradle [hunk 7]
```diff
@@ -20,13 +20,17 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-configurations {
-    jar.archiveName = 'SimpleCloud-InternalWrapper.jar'
+jar {
+    archiveFileName.set('SimpleCloud-InternalWrapper-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-internalwrapper/build.gradle:24]: Can be removed

File: simplecloud-modules/simplecloud-module-notify/build.gradle [hunk 8]
```diff
@@ -20,15 +20,19 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-configurations {
-    jar.archiveName = 'SimpleCloud-Notify.jar'
+jar {
+    archiveFileName.set('SimpleCloud-Notify-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-notify/build.gradle:24]: Can be removed

File: simplecloud-modules/simplecloud-module-npc/build.gradle [hunk 9]
```diff
@@ -20,8 +20,12 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-configurations {
-    jar.archiveName = 'SimpleCloud-NPC.jar'
+jar {
+    archiveFileName.set('SimpleCloud-NPC-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-npc/build.gradle:24]: Can be removed

File: simplecloud-modules/simplecloud-module-permission/build.gradle [hunk 10]
```diff
@@ -20,16 +20,20 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-configurations {
-    jar.archiveName = 'SimpleCloud-Permission.jar'
+jar {
+    archiveFileName.set('SimpleCloud-Permission-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-permission/build.gradle:24]: Can be removed

File: simplecloud-modules/simplecloud-module-proxy/build.gradle [hunk 11]
```diff
@@ -20,28 +20,23 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-configurations {
-    jar.archiveName = 'SimpleCloud-Proxy.jar'
+jar {
+    archiveFileName.set('SimpleCloud-Proxy-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-proxy/build.gradle:24]: Can be removed

File: simplecloud-modules/simplecloud-module-rest/build.gradle [hunk 12]
```diff
@@ -20,34 +20,31 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-configurations {
-    jar.archiveName = 'SimpleCloud-Rest.jar'
+jar {
+    archiveFileName.set('SimpleCloud-Rest-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-rest/build.gradle:24]: Can be removed

File: simplecloud-modules/simplecloud-module-serviceselection/build.gradle [hunk 13]
```diff
@@ -3,18 +3,23 @@ repositories {
...
+    testApi group: 'commons-io', name: 'commons-io', version: dependencyCommonsIOVersion
+    testApi group: 'com.google.guava', name: 'guava', version: dependencyGuavaVersion
+    testApi group: 'io.netty', name: 'netty-all', version: dependencyNettyVersion
+    testApi group: 'org.jetbrains.kotlinx', name: 'kotlinx-coroutines-core', version: dependencyKotlinCouroutinesVersion
+    testCompileOnly 'io.mockk:mockk:1.12.3'
 }
 
 configurations {
-    jar.archiveName = 'SimpleCloud-ServiceSelection.jar'
+    archivesBaseName('SimpleCloud-ServiceSelection-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-serviceselection/build.gradle:20]: Can be removed

File: simplecloud-modules/simplecloud-module-serviceselection/src/test/kotlin/eu/thesimplecloud/module/serviceselection/api/ServiceSelectionTest.kt
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-serviceselection/src/test/kotlin/eu/thesimplecloud/module/serviceselection/api/ServiceSelectionTest.kt:1]: Why was this deleted?

File: simplecloud-modules/simplecloud-module-serviceselection/src/test/kotlin/eu/thesimplecloud/module/serviceselection/api/TestServiceViewer.kt
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-serviceselection/src/test/kotlin/eu/thesimplecloud/module/serviceselection/api/TestServiceViewer.kt:1]: Why was this deleted?

File: simplecloud-modules/simplecloud-module-sign/build.gradle [hunk 14]
```diff
@@ -1,38 +1,29 @@
-dependencies {
-    implementation "org.jetbrains.kotlin:kotlin-stdlib"
-    testCompile group: 'junit', name: 'junit', version: '4.12'
+jar {
+    archiveFileName.set('SimpleCloud-Sign-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-sign/build.gradle:2]: Can be removed

File: simplecloud-modules/simplecloud-module-statistics/build.gradle [hunk 15]
```diff
@@ -20,27 +20,29 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-configurations {
-    jar.archiveName = 'SimpleCloud-Statistics.jar'
+jar {
+    archiveFileName.set('SimpleCloud-Statistics-old.jar')
```
Comment (FllipEis) [inline simplecloud-modules/simplecloud-module-statistics/build.gradle:24]: Can be removed

File: simplecloud-plugin/build.gradle [hunk 16]
```diff
@@ -21,41 +21,34 @@
...
-    compile('net.kyori:adventure-platform-bungeecord:4.1.2')
+    api('net.kyori:adventure-text-minimessage:4.11.0')
+    api('net.kyori:adventure-platform-bungeecord:4.1.2')
 }
 
 
-configurations {
-    jar.archiveName = 'SimpleCloud-Plugin.jar'
+jar {
+    archiveFileName.set('SimpleCloud-Plugin-old.jar')
```
Comment (FllipEis) [inline simplecloud-plugin/build.gradle:42]: Can be removed

File: simplecloud-runner/build.gradle [hunk 17]
```diff
@@ -8,21 +8,24 @@ repositories {
...
-    compile(group: 'eu.thesimplecloud.jsonlib', name: 'json-lib', version: "1.0.8")
+    api "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
+    testApi group: 'junit', name: 'junit', version: '4.12'
+    api(group: 'eu.thesimplecloud.jsonlib', name: 'json-lib', version: "1.0.8")
 }
 
-configurations {
-    jar.archiveName = 'runner.jar'
+jar {
+    archiveFileName.set('runner-old.jar')
```
Comment (FllipEis) [inline simplecloud-runner/build.gradle:17]: Can be removed

File: simplecloud-runner/build.gradle [hunk 18]
```diff
@@ -8,21 +8,24 @@ repositories {
...
 }
 
+shadowJar {
+    archiveFileName.set("runner.jar")
+    dependencies {
+        relocate 'kotlin', 'eu.thesimplecloud.external'
```
Comment (FllipEis) [inline simplecloud-runner/build.gradle:23]: Does this work?

**Review by FllipEis — CHANGES_REQUESTED** [review]
Why does every jar file end with `-old.jar`?

File: simplecloud-modules/simplecloud-module-serviceselection/src/test/kotlin/eu/thesimplecloud/module/serviceselection/api/TestServiceViewer.kt
Comment (MrManHD) [reply simplecloud-modules/simplecloud-module-serviceselection/src/test/kotlin/eu/thesimplecloud/module/serviceselection/api/TestServiceViewer.kt:1]: this method was not used and has no implementations so i removed it
